### PR TITLE
Add first cookbook

### PR DESCRIPTION
**Pull request by romanowski**

### CODE REVIEW

**Comment by romanowski** [comment]
I wonder what do you think about such form of cookbook? I am adding an automatic checking scripts at this moment.

File: website/docs/cookbooks/scala-versions.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# Picking Scala version with scala-cli
+
+Scala cli by default runs latest stable Scala version.
+
+Here is an unversal piece of code that detect Scala version in runtime
+
+```scala
+object ScalaVersion extends App {
```
Comment (lwronski) [inline website/docs/cookbooks/scala-versions.md:8]: I run locally this example, it throws me the following error:
```
Compiling project (Scala 3.0.2, JVM)
[error] ./ScalaVersion.scala:7:38: Not found: versions
[error]                                      ^^^^^^^^^
Error compiling project (Scala 3.0.2, JVM)
Compilation failed
```

**Review by lwronski — COMMENTED** [review]
I think, it's very valuable to create short cookbook presenting the most useful features and describing them.

File: website/docs/cookbooks/scala-versions.md (hunk 1 above)
Comment (romanowski) [reply website/docs/cookbooks/scala-versions.md:8]: Yes, that is why I added automated validation as a followup :)

File: docs_checker/check.scala [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        else 
+            val toCheck =  Files.list(path).iterator.asScala.filterNot(_.getFileName.toString.startsWith("."))
+            toCheck.toList.flatMap(checkPath)
+    catch
+        case e: Throwable =>
+            e.printStackTrace()
+            Seq(TestCase(path, Some(e)))
+
+def checkFile(file: Path) = 
+    val content = Files.lines(file).iterator.asScala.toList
```
Comment (alexarchambault) [inline docs_checker/check.scala:77]: NIT Specify a charset just in case? (`StandardCharsets.UTF_8`)

File: .github/scripts/check_docs.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+dest=$(pwd)/.scala/bin
+
+./mill -i copyTo cli.launcher $dest/scala-cli 
+
+export PATH=$dest:$PATH
+echo Adding $dest to classpath
+ls $dest
+
+./mill -i scala docs_checker/check.scala -- docs/cookbooks
```
Comment (alexarchambault) [inline .github/scripts/check_docs.sh:11]: We could make this whole file a Mill command, with something along those lines in `build.sc`:
```scala
def checkDocs(args: String*) = T.command {
  val launcher = cli.launcher().path
  os.proc(launcher, "docs_checker/check.scala", "--", if (args.isEmpty) Seq("docs/cookbooks") else args)
    .call(stdout = os.Inherit)
}
```

Run with:
```text
$ ./mill -i checkDocs
```

Watch mode should work for the Scala CLI launcher itself, when we edit its sources:
```text
$ ./mill -i -w checkDocs
```

With some clever use of Mill `T.sources`, we could even get watch mode to work for the cookbook themselves. Maybe with something like
```scala
def cookbookSources = T.sources {
  Seq(PathRef(os.pwd / "docs" / "cookbooks"))
}
def cookbookScalaSources = T.sources {
  Seq(PathRef(os.pwd / "docs_checker" / "check.scala"))
}
def checkDocs() = T.command {
  val launcher = cli.launcher().path
  val sources = cookbookSources().map(_.path)
  val scalaSources = cookbookScalaSources().map(_.path)
  os.proc(launcher, scalaSources, "--", sources)
    .call(stdout = os.Inherit)
}
```

**Review by alexarchambault — COMMENTED** [review]
I added a few comments. But they can be addressed or discussed in another PR, if we'd like to get that in sooner than later.

**Pull request merged**